marklogic-data-hub-central/ui/src/config/mocks.config.ts

```diff
@@ -1,156 +1,220 @@
-        case '/api/steps/ingestion':
-          return Promise.resolve(loadData.loads);
-        case '/api/steps/ingestion/' + loadData.loads.data[0].name + '/settings':
-          return Promise.resolve(loadData.loadSettings);
-        default:
-          return Promise.reject(new Error('not found'));
-      }
-    })
-  };
+  })
+  return axiosMock.get['mockImplementation']((url) => {
+    switch (url) {
+      case '/api/flows':
+        return Promise.resolve(loadData.flows);
+      case '/api/steps/ingestion':
+        return Promise.resolve(loadData.loads);
+      case '/api/steps/ingestion/' + loadData.loads.data[0].name:
+        return Promise.resolve(loadData.loadSettings);
+      default:
+        return Promise.reject(new Error('not found'));
+    }
+  })
+};
 
-  const curateAPI = (axiosMock) => {
-    axiosMock.delete['mockImplementation']((url) => {
-        switch (url) {
-            case '/api/steps/mapping/' + curateData.mappings.data[0].artifacts[0].name:
-                return Promise.resolve(loadData.genericSuccess);
-            default:
-                return Promise.reject(new Error('not found'))
-        }
-    });
-    axiosMock.put['mockImplementation']((url) => {
-        switch (url) {
-            case '/api/steps/mapping/' + curateData.mappings.data[0].artifacts[0].name + '/settings':
-                return Promise.resolve({
-                    "data": {},
-                    "status": 200
-                });
-            default:
-                return Promise.reject(new Error('not found'));
-        }
-    });
-    return axiosMock.get['mockImplementation']((url) => {
-      switch (url) {
-        case '/api/flows':
-          return Promise.resolve(curateData.flows)
-        case '/api/models/primaryEntityTypes':
-          return Promise.resolve(curateData.primaryEntityTypes);
-        case '/api/steps/ingestion':
-          return Promise.resolve(curateData.loads);
-        case '/api/steps/ingestion/' + loadData.loads.data[0].name + '/settings':
-          return Promise.resolve(curateData.loadSettings);
-        case '/api/steps/mapping':
-          return Promise.resolve(curateData.mappings);
-        case '/api/steps/mapping/' + curateData.mappings.data[0].artifacts[0].name + '/settings':
-          return Promise.resolve(curateData.mappingSettings);
-        case '/api/artifacts/matching':
-          return Promise.resolve(curateData.matchings);
+const curateAPI = (axiosMock) => {
+  axiosMock.delete['mockImplementation']((url) => {
+    switch (url) {
+      case '/api/steps/ingestion/' + loadData.loads.data[0].name:
+        return Promise.resolve(loadData.genericSuccess);
+      case '/api/steps/mapping/' + curateData.mappings.data[0].artifacts[0].name:
+        return Promise.resolve(loadData.genericSuccess);
+      default:
+        return Promise.reject(new Error('not found'))
+    }
+  });
+  axiosMock.post['mockImplementation']((url) => {
```

Comment:
See above, we don't use PUT now, we use POST:

https://github.com/marklogic/marklogic-data-hub/blob/feature/DHFPROD-4969-advanced-settings/marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.tsx

Reply:
The `/settings` endpoints have become redundant with these recent updates to the settings page. The majority of the properties now appear there and @rjrudin made changes to the steps endpoints to only update the properties sent. The middle-tier settings endpoint was calling the same code as the regular step endpoint.

Reply:
I am not sure what is the source of truth now to compare the PR against. The JIRA ticket here has all the endpoints.
https://project.marklogic.com/jira/browse/DHFPROD-4951
@ryanjdew is there a JIRA ticket for the recent updates to settings page you are referring to.

Reply:
I can still see PUT request when settings are saved:
<img width="1518" alt="image" src="https://user-images.githubusercontent.com/16124406/84191088-d3733280-aa4c-11ea-84b7-d4785069fa95.png">

Reply:
@bsrikan The changes came in this PR: https://github.com/marklogic/marklogic-data-hub/pull/3936 

The PUT for /settings was working exactly like the POST "/api/steps/{stepType}/{stepName}". The GET for /settings was slightly different because it only contained a subset of properties of the step.